src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -312,9 +313,23 @@ public static void ThrowInvalidOperationException_CannotSkipOnPartial()
         }
 
         [DoesNotReturn]
-        public static void ThrowInvalidOperationException_CannotMixEncodings(Utf8JsonWriter.SegmentEncoding previousEncoding, Utf8JsonWriter.SegmentEncoding currentEncoding)
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
`NoInlining` should not be applied to throw helpers, otherwise RyuJIt will not detect the callsite as cold code.